### DISCUSSION TITLE
DWI sampling scheme

### DISCUSSION THREAD

**Discussion by madrig8l** [PennLINC/qsiprep#265] [General]
Dear all,

Could someone explain the visualized DWI sampling scheme (summary file) to me in more detail? I am having difficulites understanding what is illustrated in this section. What am I looking for when comparing the original vs. post-processed scheme? 

all the best,
Bastian

**Comment by cookpa** [comment]
Do you mean the animations showing the sample points before and after processing? 

I think it's not easy to do a detailed examination of the scheme this way, but it is a quick way to tell if the scheme is roughly what is expected. A quick glance can tell us:

1. If the sampling scheme is roughly what we expect (eg a single spherical shell, multiple shells, or a cartesian grid).

2. How densely the points are sampled over the sphere. It's impossible to count exactly but you can get a general sense.

3. How uniformly the points are sampled. If they are clumped together, something is wrong.

4. How similar the corrected vs original scheme are. The normal corrections are a few degrees at most. If it's obviously different, that's a sign of problems.

**Comment by madrig8l** [comment]
Yes, that helped me understand it better. 

Regarding 1):

How would I visually discern between single / multiple shells in this illustration? It the b-value illustrated on one of the axis (labeled L, P and S?) Or is that the spatial orientation of the diffusion gradients (i.e. bvecs)? I have attached an example from my data (b=0 and b=1500, 64 directions).


![sub-01_ses-01_sampling_scheme](https://user-images.githubusercontent.com/59482591/123820700-12478500-d8fb-11eb-8bdf-8a1b36358d84.gif)

**Comment by cookpa** [comment]
It's both, the bvecs show orientation and the bvals show magnitude of the vector.

It looks like something unusual is going on. Are your b-values all approximately 1500? If so, it might be that the bvecs are not normalized to unit vectors. How did you obtain the bvec and bval files? If you'd like to upload them, I can take a look

**Comment by madrig8l** [comment]
Ok. Yes, I was wondering if there is something weird, since I expected all "dots" to be at the same distance from the center of the graph (since the magnitude is always b=1500). I have attached the files + json (converted by dcm2niix)

files:
[sub-01_ses-01_dwi.zip](https://github.com/PennLINC/qsiprep/files/6734499/sub-01_ses-01_dwi.zip)

**Comment by mattcieslak** [comment]
This may just be a strange rendering in the animation. I've seen this weird axis thing before and the bvals/vecs were ok (#59). You can see all the detailed info in the confounds.tsv file in your output directory.

**Comment by cookpa** [comment]
Thanks @mattcieslak . I checked the bvecs and they have very close to unit magnitude, as expected from dcm2niix.

**Comment by cookpa** [comment]
Both this scheme and #59 have gradient directions on a single hemisphere. I wonder if this relates to the scaling error.

**Reply by cookpa** [reply]
https://github.com/PennLINC/qsiprep/blob/cd6a9605691d0c66e746b0c36d1e9020ee0bf66d/qsiprep/interfaces/reports.py#L343-L354

I reckon it might be here, where the axis is drawn between min and max in each dimension. To enforce equal axes, one could find the global maximum over all samples and then plot axes as (-gmax, gmax) in all dimensions.
